mongo/tests/test_unit.py

```diff
@@ -145,6 +145,30 @@ def test_server_uri_sanitization(check, instance):
         assert expected_clean_name == clean_name
 
 
+def test_parse_server_config(check):
```

Comment:
Could you also add a e2e test using the new connection configuration ? 

After this https://github.com/DataDog/integrations-core/blob/1bf1bcf5f4af4a54f16bf45ad76d55c4f364ce05/mongo/tests/test_e2e.py#L47-L52

Reply:
As of today the `test_e2e` _is_ using the new connection configuration - `instance_authdb` refers to this instance:

https://github.com/DataDog/integrations-core/blob/74e71bad320547a9518dcdb3f6f4bea1ecfebc41/mongo/tests/common.py#L22-L28

Did you mean an E2E test with the legacy configuration interface?

Reply:
My bad, it seems we already cover both new and legacy.

It might worth adding an integration test with exotic characters ? Like those used in the unit test you added:

```
    instance = {
        'hosts': ['localhost', 'localhost:27018'],
        'username': 'john doe',  # Space
        'password': 'p@ss\\word',  # Special characters
        'database': 'test',
        'options': {'replicaSet': 'bar!baz'},  # Special character
    }
```

Reply:
Good idea, added one